tests/components/insteon/test_config_flow.py

```diff
@@ -0,0 +1,703 @@
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_select_hub_v1(hass: HomeAssistantType):
+    """Test we set up the Hub v1 correctly."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await _init_form(hass, HUB1)
+
+    result2, mock_setup, mock_setup_entry = await _device_form(
+        hass, result["flow_id"], mock_successful_connection, MOCK_USER_INPUT_HUB_V1
+    )
+    assert result2["type"] == "create_entry"
+    assert result2["data"] == {
+        **MOCK_USER_INPUT_HUB_V1,
+        CONF_HUB_VERSION: 1,
+    }
+
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_select_hub_v2(hass: HomeAssistantType):
+    """Test we set up the Hub v2 correctly."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await _init_form(hass, HUB2)
+
+    result2, mock_setup, mock_setup_entry = await _device_form(
+        hass, result["flow_id"], mock_successful_connection, MOCK_USER_INPUT_HUB_V2
+    )
+    assert result2["type"] == "create_entry"
+    assert result2["data"] == {
+        **MOCK_USER_INPUT_HUB_V2,
+        CONF_HUB_VERSION: 2,
+    }
+
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_failed_connection_plm(hass: HomeAssistantType):
+    """Test a failed connection with the PLM."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await _init_form(hass, PLM)
+
+    result2, _, _ = await _device_form(
+        hass, result["flow_id"], mock_failed_connection, MOCK_USER_INPUT_PLM
+    )
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_failed_connection_hub(hass: HomeAssistantType):
+    """Test a failed connection with a Hub."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await _init_form(hass, HUB2)
+
+    result2, _, _ = await _device_form(
+        hass, result["flow_id"], mock_failed_connection, MOCK_USER_INPUT_HUB_V2
+    )
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def _import_config(hass, config):
+    """Run the import step."""
+    with patch(PATCH_CONNECTION, new=mock_successful_connection,), patch(
+        PATCH_ASYNC_SETUP, return_value=True
+    ), patch(PATCH_ASYNC_SETUP_ENTRY, return_value=True):
+        return await async_import_config(hass, config)
+
+
+async def test_import_plm(hass: HomeAssistantType):
+    """Test importing a minimum PLM config from yaml."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    result = await _import_config(hass, MOCK_IMPORT_CONFIG_PLM)
+
+    assert result["type"] == "create_entry"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)